scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -459,15 +459,23 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
     """
   }
 
-  private def parseVersionAnnot(version: Tree, annot: String, field: String): Version = {
-    val versionStr = version match {
+  private def getAnnotAttribute(value: Tree): String =
+    value match {
       case x: AssignOrNamedArg => x.rhs.toString
       case x => x.toString
     }
-    try { parseVersion(versionStr) }
-    catch {
-      case NonFatal(_) => c.abort(version.pos, s"@$annot must contain $field=major.minor.patch")
+
+  private def parseVersionAnnot(version: Tree, annot: String, field: String): Version = {
+    val parsed =
+      try { parseVersion(getAnnotAttribute(version)) }
+      catch {
+        case NonFatal(_) => c.abort(version.pos, s"@$annot must contain $field=major.minor.patch")
+      }
+    majorVersion.foreach { major =>
+      if (parsed._1 < major)
+        c.abort(version.pos, s"@$annot: invalid major version (must be $major)")
```

Comment:
Shouldn't the message say something like `@$annot: version for $field no longer valid, please update` ?

Reply:
so, i don't want the version to be updated :) my goal is to make sure that all these binary-compatibility annotations are removed in a new major release (which will also result is a binary incompatibility, but that's what major releases are for). should i revise the message a bit, to say that?

Reply:
added the word `obsolete`, perhaps that makes it clearer :)

Reply:
Sounds better, thanks!